src/main/java/org/opensearch/timeseries/constant/CommonMessages.java

```diff
@@ -0,0 +1,84 @@
+    public static String EMPTY_NAME = "name should be set";
+    public static String NULL_TIME_FIELD = "Time field should be set";
+    public static String EMPTY_INDICES = "Indices should be set";
+
+    public static String getTooManyCategoricalFieldErr(int limit) {
+        return String.format(Locale.ROOT, CommonMessages.TOO_MANY_CATEGORICAL_FIELD_ERR_MSG_FORMAT, limit);
+    }
+
+    public static final String TOO_MANY_CATEGORICAL_FIELD_ERR_MSG_FORMAT = "We can have only %d categorical field/s.";
+    public static String CAN_NOT_FIND_RESULT_INDEX = "Can't find result index ";
+    public static String INVALID_CHAR_IN_RESULT_INDEX_NAME =
+        "Result index name has invalid character. Valid characters are a-z, 0-9, -(hyphen) and _(underscore)";
+    public static String FAIL_TO_VALIDATE = "fail to validate";
+    public static String INVALID_TIMESTAMP = "Timestamp field: (%s) must be of type date";
+    public static String NON_EXISTENT_TIMESTAMP = "Timestamp field: (%s) is not found in index mapping";
+    public static String INVALID_NAME = "Valid characters for name are a-z, A-Z, 0-9, -(hyphen), _(underscore) and .(period)";
+    // change this error message to make it compatible with old version's integration(nexus) test
+    public static String FAIL_TO_FIND_CONFIG_MSG = "Can't find config with id: ";
+    public static final String CAN_NOT_CHANGE_CATEGORY_FIELD = "Can't change category field";
+    public static final String CAN_NOT_CHANGE_CUSTOM_RESULT_INDEX = "Can't change custom result index";
+    public static final String CATEGORICAL_FIELD_TYPE_ERR_MSG = "A categorical field must be of type keyword or ip.";
+    // Modifying message for FEATURE below may break the parseADValidationException method of ValidateAnomalyDetectorTransportAction
+    public static final String FEATURE_INVALID_MSG_PREFIX = "Feature has an invalid query";
+    public static final String FEATURE_WITH_EMPTY_DATA_MSG = FEATURE_INVALID_MSG_PREFIX + " returning empty aggregated data: ";
+    public static final String FEATURE_WITH_INVALID_QUERY_MSG = FEATURE_INVALID_MSG_PREFIX + " causing a runtime exception: ";
+    public static final String UNKNOWN_SEARCH_QUERY_EXCEPTION_MSG =
+        "Feature has an unknown exception caught while executing the feature query: ";
+
+    // ======================================
+    // Index message
+    // ======================================
+    public static final String CREATE_INDEX_NOT_ACKNOWLEDGED = "Create index %S not acknowledged";
+    public static final String SUCCESS_SAVING_RESULT_MSG = "Result saved successfully.";
+    public static final String CANNOT_SAVE_RESULT_ERR_MSG = "Cannot save results due to write block.";
+
+    // ======================================
+    // Resource constraints
+    // ======================================
+    public static final String MEMORY_CIRCUIT_BROKEN_ERR_MSG = "AD memory circuit is broken.";
```

Comment:
"Memory consumption is too high."? But I see the other error below -- what else is involved in a circuit breaker which can be shared with an user?

Reply:
just as you said memory circuit broken means memory consumption is too high.  I changed to

```
public static final String MEMORY_CIRCUIT_BROKEN_ERR_MSG = "AD memory circuit is open due to excessive memory consumption.";
```

Reply:
This error is very close to the error in line 82. Is it the case that this error is for continually running jobs and line 82 is for the validator? Or the other way round? Then the two similar errors make sense.

Reply:
The memory circuit breaker error means the total memory usage exceeded limit, while the error in line 82 means AD memory usage exceeded limit.

Reply:
"The memory circuit breaker error means the total memory usage exceeded limit" --> why not just say that in the message? (It's fine if the error token has the same name as now.) If the user has larger machines/cluster then the total memory usage would be less of a problem and may solve their problem if this error is triggered?

Reply:
Ok. Changed to

```
public static final String MEMORY_CIRCUIT_BROKEN_ERR_MSG = "The total OpenSearch memory usage exceeds our threshold, opening the AD memory circuit. ";
```